src/util/cmdlineargs.h

```diff
@@ -67,16 +68,16 @@ class CmdlineArgs {
 
   private:
     CmdlineArgs() :
-        m_startInFullscreen(false), //Initialize vars
+        m_startInFullscreen(false), // Initialize vars
         m_midiDebug(false),
         m_developer(false),
         m_safeMode(false),
-        m_settingsPath(QDir::homePath().append("/").append(SETTINGS_PATH)) {
+        m_settingsPath(QDesktopServices::storageLocation(QDesktopServices::DataLocation).append("/")) {
```

Comment:
QDesktopServices::storageLocation(QDesktopServices::DataLocation)
returns something like ~/.local/share/data/Mixxx in Linux. 
which does not meet the XDG requirements 
I think we should use ~/.local/share/Mixxx or ~/.config/Mixxx 
~/.mixxx/analysis probably belongs to ~/.cache/Mixxx/analysis
unfortunately this destroys our --settingsPath command line feature.